crates/miden-lib/src/note/well_known_note.rs

```diff
@@ -157,4 +163,242 @@ impl WellKnownNote {
     }
+
+    /// Performs the inputs check of the provided note against the target account and the block
+    /// number.
+    ///
+    /// This function returns:
+    /// - `Some` if the note can definitely not be consumed under the provided conditions or in
+    ///   general.
+    /// - `None` if the consumption status of the note cannot be determined conclusively and further
+    ///   checks are necessary.
+    pub fn check_note_inputs(
+        &self,
+        note: &Note,
+        target_account_id: AccountId,
+        block_ref: BlockNumber,
+    ) -> Option<NoteConsumptionStatus> {
+        match self.check_note_inputs_inner(note, target_account_id, block_ref) {
+            Ok(status) => status,
+            Err(err) => {
+                let err: Box<dyn Error + Send + Sync + 'static> = Box::from(err);
+                Some(NoteConsumptionStatus::NeverConsumable(err))
+            },
+        }
+    }
+
+    /// Performs the inputs check of the provided note against the target account and the block
+    /// number.
+    ///
+    /// It performs:
+    /// - for `P2ID` note:
+    ///     - check that note inputs have correct number of values.
+    ///     - assertion that the account ID provided by the note inputs is equal to the target
+    ///       account ID.
+    /// - for `P2IDE` note:
+    ///     - check that note inputs have correct number of values.
+    ///     - check that the target account is either the receiver account or the sender account.
+    ///     - check that depending on whether the target account is sender or receiver, it could be
+    ///       either consumed, or consumed after timelock height, or consumed after reclaim height.
+    ///       See the underlying table for the specific cases. Notice that this table defines all
+    ///       possible variations of the values of the current block height, timelock height and
+    ///       reclaim height relative to each other (it was greatly reduced though).
+    /// ```text
+    ///       Where:
+    ///         - `curr` -- current block height
+    ///         - `tl`   -- timelock height
+    ///         - `rc`   -- reclaim height
+    ///         - `v`    -- symbol to show that heights have any relation
+    ///       ┏━━━┳━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━━┓
+    ///       ┃ # ┃  Height sequence   ┃   Sender    ┃  Receiver   ┃
+    ///       ┣━━━╋━━━━━━━━━━━━━━━━━━━━╋━━━━━━━━━━━━━╋━━━━━━━━━━━━━┫
+    ///       │ 1 │  rc  v  tl  ≤ curr │ return None | return None |
+    ///       ├───┼────────────────────┼─────────────┼─────────────┤
+    ///       │ 2 │ curr v  rc  <  tl  │  return tl  |  return tl  |
+    ///       ├───┼────────────────────┼─────────────┼─────────────┤
+    ///       │ 3 │ curr <  tl  ≤  rc  │  return rc  |  return tl  |
+    ///       ├───┼────────────────────┼─────────────┼─────────────┤
+    ///       | 4 │  tl  ≤ curr <  rc  │  return rc  | return None |
+    ///       └───┴────────────────────┴─────────────┴─────────────┘
```

Comment:
I find this table somewhat confusing (or at least it took me a while to understand it). It is also not exactly accurate (e.g., it is not clear what `return tl` means since we are not returning block height from this function.

I would probably replace it with textual descriptions or maybe just get rid of it as the code should be easy enough to follow.